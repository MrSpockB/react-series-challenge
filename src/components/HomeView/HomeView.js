import React, { Component } from 'react';
import {bindActionCreators} from "redux";
import { connect } from 'react-redux';
import trendingGifs from '../../actions/trendingGifs';
import {Container} from "../../App.styled";
import SearchBar from "../SearchBar/SearchBar";
import GifList from "../GifList/GifList";

class HomeView extends Component {
  constructor(props) {
    super();
    props.getTrendingGifsRequested();
  }
  render() {
    const {data} = this.props.trendingGifs;
    return (
      <Container>
        <SearchBar />
        <GifList gifs={data}/>
      </Container>
    )
  }
}

const mapStateToProps = (state) => {
  const { trendingGifs } = state;

  return { trendingGifs };
};
const mapDispatchToProps = (dispatch) => {
  const { getTrendingGifsRequested } = trendingGifs.creators;

  return bindActionCreators({ getTrendingGifsRequested }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(HomeView)