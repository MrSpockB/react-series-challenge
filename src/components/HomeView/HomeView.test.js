import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { HomeView } from "./HomeView";

describe('HomeView component', () => {
  it('should render correctly', () => {
    const output = shallow(
      <HomeView
        getTrendingGifsRequested={jest.fn}
        searchGifsClear={jest.fn}
        searchGifsRequested={jest.fn}
        trendingGifs={{data: []}}
        searchingGifs={{data: []}}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  })
});