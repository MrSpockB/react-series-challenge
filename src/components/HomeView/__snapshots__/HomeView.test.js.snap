// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView component should render correctly 1`] = `
<styled.div>
  <SearchBar
    onSearch={[Function]}
  />
  <Connect(Component)
    gifs={Array []}
  />
</styled.div>
`;
